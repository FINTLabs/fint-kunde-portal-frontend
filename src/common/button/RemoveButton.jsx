import PropTypes from "prop-types";
import React, { Component } from "react";
import Tooltip from "@material-ui/core/Tooltip";
import { IconButton, withStyles } from "@material-ui/core";
import RemoveIcon from "../../../node_modules/@material-ui/icons/CheckBox";

const styles = theme => ({
  addIcon: {
    color: theme.palette.secondary.main
  },
  removeIcon: {
    color: theme.palette.secondary.main
  }
});
class RemoveButton extends Component {
  render() {
    const { classes } = this.props;
    return (
      <Tooltip placement={this.props.placement} title={this.props.title}>
        <IconButton aria-label="Remove" onClick={this.props.onClick}>
          <RemoveIcon className={classes.removeIcon} />
        </IconButton>
      </Tooltip>
    );
  }
}

RemoveButton.defaultProps = {
  placement: "top"
};

RemoveButton.propTypes = {
  classes: PropTypes.any.isRequired,
  onClick: PropTypes.func,
  placement: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
};
export default withStyles(styles)(RemoveButton);
