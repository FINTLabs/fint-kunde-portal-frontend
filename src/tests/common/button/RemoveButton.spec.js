import React from "react";
import { create } from "react-test-renderer";
import RemoveButton from "../../../common/button/RemoveButton";
describe("Feature component", () => {
  test("it matches the snapshot", () => {
    const component = create(<RemoveButton title="test"/>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});