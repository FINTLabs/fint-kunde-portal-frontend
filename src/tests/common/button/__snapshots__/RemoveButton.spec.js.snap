// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature component it matches the snapshot 1`] = `
<button
  aria-describedby={null}
  aria-label="Remove"
  className="MuiButtonBase-root MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  title="test"
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root RemoveButton-removeIcon-2"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11H7v-2h10v2z"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;
